lib/chef/resource/hostname.rb

```diff
@@ -115,7 +119,7 @@ def updated_ec2_config_xml
 
           # make sure node['fqdn'] resolves via /etc/hosts
           unless new_resource.ipaddress.nil?
-            newline = "#{new_resource.ipaddress} #{new_resource.hostname}"
+            newline = "#{new_resource.ipaddress} #{new_resource.fqdn} #{new_resource.hostname}"
```

Comment:
This approach will introduce an extra space for all existing users that are not using the FQDN. You need to ensure the formatting doesn't change for those users